cmd/export/discover.go

```diff
@@ -0,0 +1,214 @@
+package export
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/pager"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+)
+
+// groupResource contains the APIGroup and APIResource
+type groupResource struct {
+	APIGroup        string
+	APIVersion      string
+	APIGroupVersion string
+	APIResource     metav1.APIResource
+	objects         *unstructured.UnstructuredList
```

Comment:
nit: either make all the fields unexported or non. either way the type is unexported so I don't see a need to mix it in this type.

Reply:
+1, will follow up